_posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md

```diff
@@ -0,0 +1,78 @@
+---
+title: New API Blueprint Tutorial and Examples
+excerpt: We've introduced a brand new API Blueprint tutorial along with some new examples.
+layout: post
+date: 2015-03-03 16:00:00 +0000
+author: kyle
+published: true
+comments: true
+---
+
+We've introduced a new [API Blueprint tutorial](http://apiary.io/blueprint) along with some brand new [API Blueprint examples](http://docs.pollsapi.apiary.io/). Our new tutorial walks you though the basics of API Blueprint along with how you can build your very first blueprint. It uses an example of a polls service, which allows consumers to view polls and vote in them.
+
+There is an [example implementation](http://github.com/apiaryio/polls-api) of the Polls API so you can perform requests to a real API.
+
+```bash
+$ curl https://polls.apiblueprint.org/questions/1
+```
+```json
+{
+    "question": "Favourite programming language?",
+    "published_at": "2014-11-11T08:40:51.620Z",
+    "url": "/questions/2",
+    "choices": [
+        {
+            "choice": "Swift",
+            "url": "/questions/1/choices/1",
+            "votes": 0
+        }, {
+            "choice": "Python",
+            "url": "/questions/1/choices/2",
+            "votes": 0
+        }
+    ]
+}
+```
+
+We've deployed an instance of this API and it's accessible for anyone to experiment with: `https://polls.apiblueprint.org/`.
+
+You can also deploy your own instance of the Polls API to Heroku with a single click:
+
+[![Deploy to Heroku](https://www.herokucdn.com/deploy/button.png)](https://heroku.com/deploy?template=https://github.com/apiaryio/polls-api)
+
+The Polls API is also a [hypermedia](TODO) API supporting the [HAL](http://stateless.co/hal_specification.html) and [Siren](https://github.com/kevinswiber/siren) hypermedia content-types based on [content negotiation](https://github.com/apiaryio/api-blueprint/blob/master/examples/6.%20Requests.md). This is demonstrated in an advanced version of the Polls API [example blueprint](http://docs.pollshypermedia.apiary.io/).
```

Comment:
Also, this is a really great introduction:
http://oredev.org/2010/sessions/hypermedia-apis

Also, you might look at:

http://www.infoq.com/articles/hypermedia-api-tutorial-part-one
http://www.infoq.com/articles/hypermedia-api-tutorial-part-two

On Mon, Mar 2, 2015 at 5:32 PM, Mark Foster <mark@apiary.io> wrote:

> > I'd like to point readers to beginners information on what hypermedia
> in the hypermedia hyperlink.
>
> @kylef I don't understand this question. Are you saying you have an empty
> hyperlink you want to connect to something?
>
> On Mon, Mar 2, 2015 at 5:14 PM, Kyle Fuller <notifications@github.com>
> wrote:
>
>> In _posts/2015-03-03-Polls-API-Blueprint-Tutorial-Examples.md
>> <https://github.com/apiaryio/apiaryio.github.com/pull/40#discussion_r25655708>
>> :
>>
>> > +        }, {
>> > +            "choice": "Python",
>> > +            "url": "/questions/1/choices/2",
>> > +            "votes": 0
>> > +        }
>> > +    ]
>> > +}
>> > +```
>> > +
>> > +We've deployed an instance of this API and it's accessible for anyone to experiment with: `https://polls.apiblueprint.org/` <https://polls.apiblueprint.org/>.
>> > +
>> > +You can also deploy your own instance of the Polls API to Heroku with a single click:
>> > +
>> > +[![Deploy to Heroku](https://www.herokucdn.com/deploy/button.png)](https://heroku.com/deploy?template=https://github.com/apiaryio/polls-api)
>> > +
>> > +The Polls API is also a [hypermedia](TODO) API supporting the [HAL](http://stateless.co/hal_specification.html) and [Siren](https://github.com/kevinswiber/siren) hypermedia content-types based on [content negotiation](https://github.com/apiaryio/api-blueprint/blob/master/examples/6.%20Requests.md). This is demonstrated in an advanced version of the Polls API [example blueprint](http://docs.pollshypermedia.apiary.io/).
>>
>> I'd like to point readers to beginners information on what hypermedia in
>> the hypermedia hyperlink. Does anyone have any suggestions of a decent
>> resource?
>>
>> /c @smizell <https://github.com/smizell>
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/apiaryio/apiaryio.github.com/pull/40/files#r25655708>
>> .
>>
>
>